woodwork/statistics_utils.py

```diff
@@ -640,3 +647,39 @@ def _get_histogram_values(series, bins=10):
         )
 
     return results
+
+
+def _infer_temporal_frequencies(dataframe, temporal_columns=None):
+    """Infers the observation frequency (daily, biweekly, yearly, etc) of each temporal column
+            in the DataFrame. Temporal columns are ones with the logical type Datetime or Timedelta.
+
+    Args:
+        dataframe (DataFrame): The DataFrame for which frequncies should be inferred.
+        temporal_columns (list[str], optional): Columns for which frequencies should be inferred. Must be columns
+            that are present in the DataFrame and are temporal in nature. Defaults to None. If not
+            specified, all temporal columns will have their frequencies inferred.
+
+    Returns:
+        (dict): A dictionary where each key is a temporal column from the DataFrame, and the
+            value is its observation frequency represented as a pandas offset alias string (D, M, Y, etc.)
+            or None if no uniform frequency was present in the data.
+    """
+    logical_types = dataframe.ww.logical_types
+
+    if temporal_columns is None:
+        temporal_columns = [
+            col
+            for col, ltype in logical_types.items()
+            if isinstance(ltype, (Datetime, Timedelta))
+        ]
+    else:
+        for col in temporal_columns:
+            if col not in dataframe:
+                raise ValueError(f"Column {col} not found in dataframe.")
+            ltype = logical_types[col]
+            if not isinstance(ltype, (Datetime, Timedelta)):
```

Comment:
I'm curious why we're not including PeriodIndex here?

`period_range = pd.period_range(start='1/1/21', freq="3D", periods=4)`
`print(period_range)`
`print(period_range.freq)`
`>>> PeriodIndex(['2021-01-01', '2021-01-04', '2021-01-07', '2021-01-10'], dtype='period[3D]')`
`>>> <3 * Days>`

Reply:
These are Woodwork logical types! Any column that's in a woodwork dataframe will be a pandas Series, and the only pandas dtypes Woodwork allows that are valid inputs to `infer_freq` are the ones attached to these logical types

Reply:
Ahh completely misread those, awesome looks perfect!